cellprofiler_core/readers/ngff_reader.py

```diff
@@ -0,0 +1,340 @@
+import numpy
+import zarr
+
+from ..constants.image import MD_SIZE_S, MD_SIZE_C, MD_SIZE_Z, MD_SIZE_T, MD_SIZE_Y, MD_SIZE_X, MD_SERIES_NAME
+from ..constants.reader import ZARR_FILETYPE
+
+from ..reader import Reader
+
+import os
+
+import re
+import logging
+from lxml import etree
+
+
+FORMAT_TESTER = re.compile(r"\.zarr([\\/]|$)", flags=re.IGNORECASE)
+
+LOGGER = logging.getLogger(__name__)
+
+
+class NGFFReader(Reader):
+    """ Derive from this abstract Reader class to create your own image reader in Python
+
+    You need to implement the methods below in the derived class.
+    """
```

Comment:
Should change to something like (I need to do this on the Bio-Formats reader as well):

```
    """
    Reader for zarr files, specifically those following the OME-NGFF file format,
    although has some support for other zarr image files, depending on their contents.
    """
```

Reply:
Well spotted, thanks!